app/main.py

```diff
@@ -1,6 +1,15 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    uniques_names_functions = {}
+
+    def inner(*args) -> Any:
+        if args not in uniques_names_functions:
+            result_function = func(*args)
```

Comment:
Your approach is incorrect: you should store arguments and the result of the function with these arguments(As a key of dict).